### PR TITLE
Dsdeepb 290 task config management

### PR DESCRIPTION
**Pull request by peilin**
vanilla version of method configuration CRUD, version has not been implemented yet.

### CODE REVIEW

File: src/main/scala/org/broadinstitute/dsde/rawls/model/WorkspaceModel.scala [hunk 1]
```diff
@@ -69,6 +69,40 @@ case class Entity(
...
+case class Method(
+                   @(ApiModelProperty@field)(required = true, value = "The name of the method")
+                   name: String,
+                   @(ApiModelProperty@field)(required = true, value = "The namespace of the method")
+                   nameSpace: String,
+                   @(ApiModelProperty@field)(required = true, value = "The version of the method")
+                   version: String
+                 )
+@ApiModel(value = "Method Configuration")
+case class MethodConfiguration(
```
Comment (dvoet) [inline src/main/scala/org/broadinstitute/dsde/rawls/model/WorkspaceModel.scala]: the method configs themselves should have a namespace in addition to that of the workspace, I meant you should add that namespace.

File: src/main/scala/org/broadinstitute/dsde/rawls/webservice/RawlsApiService.scala [hunk 2]
```diff
@@ -241,4 +284,107 @@ trait WorkspaceApiService extends HttpService with PerRequestCreator {
...
+      post {
+        entity(as[MethodConfiguration]) { methodConfiguration =>
+          requestContext => perRequest(requestContext, WorkspaceService.props(workspaceServiceConstructor),
+            WorkspaceService.CreateMethodConfiguration(workspaceNamespace, workspaceName, methodConfiguration))
+        }
+      }
+    }
+  }
+
+  @Path("/{workspaceNamespace}/{workspaceName}/methodconfigs")
```
Comment (dvoet) [inline src/main/scala/org/broadinstitute/dsde/rawls/webservice/RawlsApiService.scala]: path does not match

File: src/main/scala/org/broadinstitute/dsde/rawls/model/WorkspaceModel.scala (hunk 1 above)
Comment (peilin) [inline src/main/scala/org/broadinstitute/dsde/rawls/model/WorkspaceModel.scala]: I used case class WorkspaceName, which has the namespace

Comment (dvoet) [inline src/main/scala/org/broadinstitute/dsde/rawls/model/WorkspaceModel.scala]: workspaces have their own namespace which is different than the method config namespace. Add namespace to this class. Then all urls need to be changes as well, e.g. to get a method config would be
/{workspaceNamespace}/{workspaceName}/methodconfigs/{methodConfigurationNamespace}/{methodConfigurationName}

once this is done you can merge it

Comment (peilin) [inline src/main/scala/org/broadinstitute/dsde/rawls/model/WorkspaceModel.scala]: just saw this, yeah, that makes sense

**Pull request merged**